influxql/ast.go

```diff
@@ -3988,7 +3988,7 @@ func evalBinaryExpr(expr *BinaryExpr, m map[string]interface{}) interface{} {
 				if !ok {
 					return nil
 				} else if rhs == 0 {
-					return float64(0)
+					return int64(0)
```

Comment:
So this is a lot more confusing than I thought it would be. We appear to have an inconsistency. If you do division of two integers in a condition, you get integer division. I verified that the division below is between two integers. But, if you do division between two integers in a field (binary math), you get float division and will get a float (which EvalType doesn't return correctly and so would screw up subqueries).

I think we need to straighten out this behavior and backport that specific section to the 1.3 series.

CC @rbetts